### PR TITLE
dont offer to edit external config settings if we can't edit them

### PR DESCRIPTION
**Pull request by icewind1991**
no need to confuse the user by providing options that he cant use.

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

**Comment by juliushaertl** [comment]
Rebased and pushed a fix for the tests

**Review by juliusknorr — APPROVED** [review]

File: apps/files_external/js/statusmanager.js [hunk 1]
```diff
@@ -182,12 +183,14 @@ OCA.Files_External.StatusManager = {
...
 						this.showCredentialsDialog(name, mountData);
-					} else {
+					} else if (mountData.canEdit) {
 						OC.dialogs.confirm(t('files_external', 'There was an error with message: ') + mountData.error + '. Do you want to review mount point config in admin settings page?', t('files_external', 'External mount error'), function (e) {
 							if (e === true) {
 								OC.redirect(OC.generateUrl('/settings/admin/externalstorages'));
 							}
 						});
+					} else {
+						OC.dialogs.info(t('files_external', 'There was an error with message: ') + mountData.error + '. Please contact your system administrator.', t('files_external', 'External mount error'), () => {});
```
Comment (MorrisJobke) [inline apps/files_external/js/statusmanager.js:193]:
```suggestion
						OC.dialogs.info(t('files_external', 'There was an error with message: ') + mountData.error + '. Please contact your system administrator.', t('files_external', 'External mount error'), function() {});
```

Comment (MorrisJobke) [reply apps/files_external/js/statusmanager.js:193]: We still support IE which does not support the short form.

**Comment by MorrisJobke** [comment]
CI fails.

**Comment by icewind1991** [comment]
CI is happy (except for our scheduled mysql test timeout)

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**